man/froll.Rd

```diff
@@ -45,36 +49,70 @@ frollapply(x, n, FUN, \dots, fill=NA, align=c("right", "left", "center"))
   conveniently within \code{data.table} syntax.
 
   Argument \code{n} allows multiple values to apply rolling functions on
-  multiple window sizes. If \code{adaptive=TRUE}, then \code{n} must be a list.
-  Each list element must be integer vector of window sizes corresponding
-  to every single observation in each column; see Examples.
+  multiple window sizes. If \code{adaptive=TRUE}, then \code{n} must be a list,
+  see \emph{Adaptive rolling functions} section below for details.
 
-  When \code{algo="fast"} an \emph{"on-line"} algorithm is used, and
-  all of \code{NaN, +Inf, -Inf} are treated as \code{NA}.
-  Setting \code{algo="exact"} will make rolling functions to use a more
-  computationally-intensive algorithm that suffers less from floating point
-  rounding error (the same consideration applies to \code{\link[base]{mean}}).
-  \code{algo="exact"} also handles \code{NaN, +Inf, -Inf} consistently to
-  base R. In case of some functions (like \emph{mean}), it will additionally
+  When multiple columns or multiple window widths are provided, then they
+  are run in parallel. The exception is for \code{algo="exact"}, which runs in
+  parallel already. By default data.table uses only half of available CPUs,
+  see \code{\link{setDTthreads}} for details.
+}
+\section{\code{hasNA} argument}{
+  \code{hasNA} can be used to speed up processing in cases when it is known
+  if \code{x} contains \code{NA, NaN, +Inf, -Inf}.
+  \itemize{
+    \item{ Default \code{hasNA=NA} will use faster NA inaware implementation,
+    but when NAs are detected it will re-run NA aware implementation. }
+    \item{ \code{hasNA=TRUE} will use NA aware implementation straightaway. }
+    \item{ \code{hasNA=FALSE} will use faster NA inaware implementation.
+    Then depending on the rolling function it will either
+    \itemize{
+      \item{ (\emph{mean, sum}) detect NAs, raise warning, re-run NA aware. }
+      \item{ (\emph{max}) not detect NAs and may silently produce incorrect
+      answer. }}
```

Comment:
```suggestion
    \item{ \code{hasNA=TRUE} will use \code{NA} aware implementation straightaway. }
    \item{ \code{hasNA=FALSE} will use faster \code{NA} agnostic implementation.
    Then depending on the rolling function it will either
    \itemize{
      \item{ (\emph{mean, sum}) detect \code{NA}s, raise warning, re-run \code{NA} aware. }
      \item{ (\emph{max}) not detect \code{NA}s and may silently produce an incorrect
      answer. }}
```